# Week 2 Checkpoint: First Data Profile
**BAN 6003 Data Management and Analytics Integration**

Complete this notebook using the `employee_data.csv` file in the `data/` folder.

This is the independent checkpoint for Week 2. Use the functions practiced in the guided lab. You are not expected to clean the data yet. Your goal is to profile the data and report what you notice.


## Instructions

1. Run each cell from top to bottom.
2. Fill in all cells marked `# Your code here`.
3. Write short answers in the markdown response areas.
4. Save, commit, and push your notebook to GitHub.

You may use the Week 2 guided lab as a reference.


## Task 0: Import Pandas and Load the Data
Run the cell below.


In [15]:
import pandas as pd

df = pd.read_csv("../data/employee_data.csv")
df.head()

,employee_id,first_name,last_name,hire_date,department,salary,age,full_time
0,101,John,Smith,4/15/22,Marketing,85000,34.0,True
1,102,Jane,Doe,8/20/21,Sales,92000,45.0,True
2,103,Peter,Jones,1/2/22,HR,71000,29.0,True
3,104,Mary,Johnson,11/10/23,Engineering,115000,38.0,True
4,105,Mike,Brown,5/30/20,sales,89000,NaN,False


## Task 1: Basic Structure
Show the basic structure of the DataFrame.

Your code should show:
- the shape of the DataFrame;
- the column names;
- the first five rows;
- the last five rows.


In [22]:
print("shape", df.shape)
print("columns", df.columns)
display(df.head())
display(df.tail())


shape (13, 8)
columns Index(['employee_id', 'first_name', 'last_name', 'hire_date', 'department',
       'salary', 'age', 'full_time'],
      dtype='object')


,employee_id,first_name,last_name,hire_date,department,salary,age,full_time
0,101,John,Smith,4/15/22,Marketing,85000,34.0,True
1,102,Jane,Doe,8/20/21,Sales,92000,45.0,True
2,103,Peter,Jones,1/2/22,HR,71000,29.0,True
3,104,Mary,Johnson,11/10/23,Engineering,115000,38.0,True
4,105,Mike,Brown,5/30/20,sales,89000,NaN,False


,employee_id,first_name,last_name,hire_date,department,salary,age,full_time
8,102,Emily,White,1/15/24,Sales,91000,31.0,True
9,110,David,Miller,7/22/19,HR,68000,58.0,True
10,111,Sarah,Moore,3/12/23,Marketing,"$91,000",28.0,True
11,error,Kevin,Clark,2/5/24,Sales,88000,35.0,True
12,113,Brian,Hall,6/18/22,Engineering,5900000,999.0,True


### Task 1 Written Response
Write 1–2 sentences describing the size and general structure of the dataset.

**Your response:**  
[The dataset consists of 13 rows (employee records) and 8 columns. It contains general employee information including their ID, name, hire date, department, salary, age, and employment status.]


## Task 2: Data Types and Non-Missing Counts
Use Pandas to inspect data types and non-missing counts.

Your code should use:
- `info()`
- `dtypes`


In [28]:
print("Data Types:")
display(df.dtypes)

print("\nDataFrame Info:")
df.info()

Data Types:


employee_id     object
first_name      object
last_name       object
hire_date       object
department      object
salary          object
age            float64
full_time       object
dtype: object


DataFrame Info:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13 entries, 0 to 12
Data columns (total 8 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   employee_id  13 non-null     object 
 1   first_name   13 non-null     object 
 2   last_name    13 non-null     object 
 3   hire_date    12 non-null     object 
 4   department   13 non-null     object 
 5   salary       12 non-null     object 
 6   age          12 non-null     float64
 7   full_time    12 non-null     object 
dtypes: float64(1), object(7)
memory usage: 960.0+ bytes


### Task 2 Written Response
Which columns, if any, have data types that might need to be investigated later? For example, is a number or date stored as text/object?

**Your response:**  
The hire_date and salary columns are currently stored as object (text) types. They will need to be investigated and converted to datetime and numeric data types, respectively, for proper analysis.

## Task 3: Missing Values
Check missing values for all columns.

Your code should use `isna().sum()`.

Optional: also calculate missing percentages.


In [33]:
print("Missing Values Count:")
display(df.isna().sum())

print("\nMissing Values Percentage (%):")
display((df.isna().sum() / len(df)) * 100)

Missing Values Count:


employee_id    0
first_name     0
last_name      0
hire_date      1
department     0
salary         1
age            1
full_time      1
dtype: int64


Missing Values Percentage (%):


employee_id    0.000000
first_name     0.000000
last_name      0.000000
hire_date      7.692308
department     0.000000
salary         7.692308
age            7.692308
full_time      7.692308
dtype: float64

### Task 3 Written Response
Which columns have missing values? Which missing values seem most important to investigate later?

**Your response:**  
The hire_date, salary, age, and full_time columns each have 1 missing value (approximately 7.7% of the dataset). The missing salary and hire_date values are the most important to investigate, as they are critical components for payroll and HR analytics.


## Task 4: Numerical Summary
Use `.describe()` to summarize numerical columns.

Then choose one numerical column and calculate at least two individual summary statistics such as mean, median, min, max, or standard deviation.


In [37]:
print("Numerical Summary:")
display(df.describe())

print("\nAge Mean (평균 나이):", df['age'].mean())
print("Age Max (최대 나이):", df['age'].max())

Numerical Summary:


,age
count,12.000000
mean,118.583333
std,277.411889
min,28.000000
25%,32.500000
50%,36.500000
75%,46.750000
max,999.000000



Age Mean (평균 나이): 118.58333333333333
Age Max (최대 나이): 999.0


### Task 4 Written Response
What do you notice from the numerical summaries? Mention at least one range, unusual value, or pattern.

**Your response:**  
The numerical summary for the age column reveals an extreme outlier: a maximum age of 999.0. This highly unusual value severely skews the mean age to approximately 118.6 years, indicating a data entry error or placeholder that must be cleaned before further analysis.

## Task 5: Categorical Summary
Use `.value_counts()` to summarize at least one categorical column.

Start with `department` if it exists in the dataset. Then choose one additional categorical column if available.


In [40]:
print("Department Counts:")
display(df['department'].value_counts())

print("\nFull Time Status Counts:")
display(df['full_time'].value_counts())

Department Counts:


department
Sales          4
Marketing      3
Engineering    3
HR             2
sales          1
Name: count, dtype: int64


Full Time Status Counts:


full_time
True     11
False     1
Name: count, dtype: int64

### Task 5 Written Response
What are the most common categories? Do you notice anything that might need standardization or further investigation later?

**Your response:**  
The most common department is Sales, and the vast majority of employees are full-time (11 True vs 1 False). However, the department column requires standardization, as "Sales" and "sales" are currently being counted as two separate categories due to inconsistent capitalization.

## Task 6: Duplicate Checks
Check for duplicate records and duplicate employee IDs.

Your code should check:
- fully duplicated rows;
- duplicate `employee_id` values;
- the most common `employee_id` values using `value_counts()`.


In [42]:
print("Fully Duplicated Rows:", df.duplicated().sum())

print("\nDuplicate Employee IDs Count:", df['employee_id'].duplicated().sum())

print("\nMost Common Employee IDs:")
display(df['employee_id'].value_counts().head(3))

Fully Duplicated Rows: 0

Duplicate Employee IDs Count: 1

Most Common Employee IDs:


employee_id
102    2
101    1
103    1
Name: count, dtype: int64

### Task 6 Written Response
Are there duplicate employee IDs? If yes, what might that mean if `employee_id` is supposed to uniquely identify each employee?

**Your response:**  
Yes, there is one duplicate employee ID (ID 102 appears twice). Since there are no fully duplicated rows, this indicates that either one employee's record was entered twice with slight variations, or two different employees were mistakenly assigned the exact same ID.

## Task 7: Overall Data Quality Observations
Write 3–5 bullet points summarizing the condition of the dataset.

Focus on what you found from profiling. Do **not** clean the data yet.

Examples:
- The dataset has ___ rows and ___ columns.
- The ___ column has missing values.
- The ___ column may have an incorrect data type.
- The ___ column has possible duplicate IDs.
- The ___ column has a range from ___ to ___.


**Your observations:**

1. The dataset consists of 13 rows and 8 columns, with 1 missing value each in the hire_date, salary, age, and full_time columns.
2. The salary and hire_date columns have incorrect data types (object) and must be converted to numeric and datetime formats.
3. There is a severe outlier in the age column (maximum age of 999.0) indicating a clear data entry error.
4. The department column requires standardization due to inconsistent capitalization ("Sales" vs "sales").
5. There is one duplicate employee_id (ID 102) that needs to be investigated and resolved to ensure unique identification.


## Submission Checklist
Before submitting, make sure:

- [ ] All code cells run without errors.
- [ ] You completed all tasks.
- [ ] You wrote short responses in the markdown cells.
- [ ] You saved the notebook.
- [ ] You committed and pushed your work to GitHub.

Use the terminal:

```bash
git add .
git commit -m "Completed Week 2 data profile checkpoint"
git push
```
